src/portal-components/DetailsPage/SideNavMenu.tsx

```diff
@@ -63,7 +63,7 @@ export const SideNavMenu: React.FC<{
               onClick={
                 isDisabled
                   ? undefined
-                  : () => handleMenuClick(getComponentId(el))
+                  : () => handleMenuClick(getComponentId(el, '', index))
```

Comment:
I don't like that we're using the index for the ID. It will break links to sections if content is added/reorganized within the section. But without the index, we may have collisions. Maybe each section should require a unique, human-readable identifier?